### PR TITLE
[Mellanox] Change NVIDIA SDK build to cmake method

### PR DESCRIPTION
**Pull request by dprital**
#### Why I did it

Up to now, building Mellanox SDK debian packages was done by "autotools" method.
Lately, Mellanox SDK team added the support to build by "cmake" method.
The benefit of using "cmake" is with decrease time of SDK build as "cmake" make it faster then "auto build".

When moving to "cmake" method, all SDK debians:

* applibs
* sx-complib
* sx-example
* sx-gen-utils
* sx-scew
* sxd-libs
* python-sdk-api
* sx-acl-helper
* wjh-libs
* sx-hash-calc
* sx-obj-desc-lib
* sxd-libs

will be replaced by single debian - sys_sdk.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

* Update cmake version on sonic slave dockers (bookworm and bullseye)
* Change SDK Makefiles, delete all Makefiles of removed debians and add new Makefile for building sys_sdk.
* Change sdk.mk and sdk.dep acordingly.
* sx-kernel package was remain since it is consumed by the host as well and I didn't want to consume sys_sdk in host.

#### How to verify it

* Enter syncd docker and verify the list of SDK packages are sx-kernel and sys-sdk only
* Run regression tests and verify there is no degradation
* Install the debian that include dbg-symbols and verify there are debug symbols by setting relevant breakpoints
* Test the build time when using "cmake" for building SDK packages, compare to "auto build"

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by stepanblyschak — APPROVED** [review]

**Review by liat-grozovik — DISMISSED** [review]

**Comment by bingwang-ms** [comment]
The SDK/SAI change LGTM. Need build team to review the cmake related change. @liushilongbuaa Can you please help review?

**Comment by bingwang-ms** [comment]
@dprital I suggest separating the PR into two PRs, one for SAI/SDK change, and the other one for cmake related change. Please let me know how do you think. Thanks

**Comment by bingwang-ms** [comment]
@dprital Can you remove the SAI/SDK change from this PR since we already have https://github.com/sonic-net/sonic-buildimage/pull/20420 ?

File: sonic-slave-bullseye/Dockerfile.j2 [hunk 1]
```diff
@@ -458,6 +458,14 @@ RUN eatmydata apt install -y \
         libabsl20200923 libc-ares2 python3-six libboost-thread1.74.0 libboost-dev libboost-system-dev libboost-thread-dev libboost-filesystem1.74.0 libboost-program-options1.74.0 libboost-thread1.74.0 libnanomsg5 libpcap0.8 libthrift-0.13.0 libboost-dev libboost-filesystem-dev libboost-program-options-dev libgmp-dev libnanomsg-dev libpcap-dev libtool pkg-config libthrift-dev python3-thrift thrift-compiler libboost-iostreams1.74.0 libgc1 cpp libboost-dev libboost-all-dev libboost-graph-dev libboost-iostreams-dev libfl-dev libgc-dev libgmp-dev libbpf-dev tcpdump libelf-dev llvm clang python3-pyroute2 python3-ply python3-scapy python3-setuptools python3-thrift libthrift-0.13.0
 {%- endif %}
 
+{%- if CONFIGURED_ARCH == "amd64" %}
+# Upgrade CMAKE version
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.27.6/cmake-3.27.6-linux-x86_64.sh
```
Comment (liushilongbuaa) [inline sonic-slave-bullseye/Dockerfile.j2:463]: How about apt-get install?
Is there any version dependency?

**Comment by dprital** [comment]
> @dprital Can you remove the SAI/SDK change from this PR since we already have #20420 ?

Done

**Pull request merged**